### PR TITLE
Add filtering to BBH exec for local or global time stepping

### PR DESCRIPTION
**Pull request by geoffrey4444**
## Proposed changes

Currently, filtering is only added in the BBH exec for global time stepping. This PR adds it whether you use global or local time stepping.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

**Comment by geoffrey4444** [comment]
Test failure is a timeout that looks unrelated to this PR to me

File: tests/InputFiles/GeneralizedHarmonic/BinaryBlackHole.yaml [hunk 1]
```diff
@@ -157,6 +157,13 @@ SpatialDiscretization:
     Formulation: StrongInertial
     Quadrature: GaussLobatto
 
+Filtering:
+  ExpFilter0:
+    Alpha: 36.0
+    HalfPower: 210
```
Comment (kidder) [inline tests/InputFiles/GeneralizedHarmonic/BinaryBlackHole.yaml]: this seems a bit extreme...

Comment (geoffrey4444) [reply tests/InputFiles/GeneralizedHarmonic/BinaryBlackHole.yaml]: Do you have a value you'd prefer? Note the filter is disabled by default anyway, and in real life the user will almost certainly need to change this depending on how they change the resolution

Comment (kidder) [reply tests/InputFiles/GeneralizedHarmonic/BinaryBlackHole.yaml]: how about 24 or 32...

**Review by knelli2 — DISMISSED** [review]

**Review by kidder — DISMISSED** [review]

**Comment by kidder** [comment]
please squash and rebase

**Review by knelli2 — APPROVED** [review]

**Review by kidder — APPROVED** [review]

**Pull request merged**